R/pstr.R

```diff
@@ -49,11 +45,19 @@ pstr_at_product_level <- function(companies, scenarios, low_threshold = 1 / 3, h
 
   .companies |>
     xstr_add_values_to_categorize(.scenarios) |>
-    add_risk_category(low_threshold, high_threshold, .default = NA) |>
+    xstr_add_risk_category(.default = NA) |>
     xstr_polish_output_at_product_level() |>
     pstr_select_cols_at_product_level()
 }
 
+# Threshold categories based on years to segment low, medium, and high reduction targets.
+threshold_categories <- function(data) {
+  case_when(
+    data == 2030 ~ list(low = 1 / 9, high = 2 / 9),
+    .default = list(low = 1 / 3, high = 2 / 3)
+  )
+}
```

Comment:
This is a very good idea! I think we can simplify it a bit with `ifelse()` and expose it as the new default of `low_threshold` and `high_threshold`.